api4/team.go

```diff
@@ -215,6 +216,34 @@ func updateTeamMemberRoles(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func updateTeam(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+	
+	team := model.TeamFromJson(r.Body)
+
+	if team == nil {
+		c.SetInvalidParam("team")
+		return
+	}
```

Comment:
You can't actually change the `id` of a team so it's fine to only pass in the team for updating and use the `id` on the team to construct the path. Can you switch it so the driver function only takes in a team? You can still test the endpoint by hitting it directly with `Client.DoApiPost`, rather than going through the driver function